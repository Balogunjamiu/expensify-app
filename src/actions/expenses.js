
import {  v4 as uuidv4} from 'uuid'
//Add Expenses 
export const addExpense = ({
    description = '',
     note = '',
     amount = 0,
     createdAt = 0
    }={})=>({
    type: 'ADD_EXPENSE',
    expense : {
        id: uuidv4(),
        description,
        note,
        amount,
        createdAt
    }
})
 export const removeExpense = ({id}={}) =>({
    type: 'REMOVE_EXPENSE',
    id 
})
 export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
})
