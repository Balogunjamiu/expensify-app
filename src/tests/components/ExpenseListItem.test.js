import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses'
import React from 'react'
import {shallow} from 'enzyme';


test('should render expenseplist item with fixtures', ()=>{
    const wrapper = shallow(<ExpenseListItem {...expenses[1]}/>)
    expect(wrapper).toMatchSnapshot()

})
