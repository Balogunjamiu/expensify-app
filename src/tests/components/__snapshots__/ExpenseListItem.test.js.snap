// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenseplist item with fixtures 1`] = `
<div>
  <Link
    to="/edit/2"
  >
    <h3>
      Rent
    </h3>
  </Link>
  <p>
    $1,095.00
    -
    December 28th, 1969
  </p>
</div>
`;
